File: Java/백준/Bronze/2975. Transactions/Transactions.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line;

        while (true) {
            line = br.readLine();
            String[] parts = line.split(" ");
            int balance = Integer.parseInt(parts[0]);
            String type = parts[1];
            int amount = Integer.parseInt(parts[2]);

            if (balance == 0 && type.equals("W") && amount == 0) {
                break;
            }

            if (type.equals("W")) {
                if (balance - amount < -200) {
                    System.out.println("Not allowed");
                } else {
                    balance -= amount;
                    System.out.println(balance);
                }
            } else if (type.equals("D")) {
                balance += amount;
                System.out.println(balance);
            }
        }
    }
}

File: C11/백준/Bronze/2975. Transactions/Transactions.c
#include <stdio.h>

int main() {
    int balance, amount;
    char type;

    while (1) {
        scanf("%d %c %d", &balance, &type, &amount);
        if (balance == 0 && type == 'W' && amount == 0) {
            break;
        }

        if (type == 'W') {
            if (balance - amount < -200) {
                printf("Not allowed\n");
            } else {
                balance -= amount;
                printf("%d\n", balance);
            }
        } else if (type == 'D') {
            balance += amount;
            printf("%d\n", balance);
        }
    }

    return 0;
}